src/demo-app/datepicker/datepicker-demo.ts

```diff
@@ -40,4 +41,49 @@ export class DatepickerDemo {
 
   onDateInput = (e: MatDatepickerInputEvent<Date>) => this.lastDateInput = e.value;
   onDateChange = (e: MatDatepickerInputEvent<Date>) => this.lastDateChange = e.value;
+
+  // pass custom header component type as input
+  customHeader = CustomHeader;
+}
+
+// Custom header component for datepicker
+@Component({
+  selector: 'custom-header',
+  template: `
+      <div>
```

Comment:
this template is big enough that I would split it into a separate file

Reply:
done in 5f4b0fc435cc1e88f2ba708c66fc7fc48340a60c